Content.Server/StationEvents/Events/SolarFlare.cs

```diff
@@ -33,31 +36,25 @@ public override void Added()
         _event.EndAfter = RobustRandom.Next(ev.MinEndAfter, ev.MaxEndAfter);
     }
 
-    public override void Started()
-    {
-        base.Started();
-        MessLights();
-    }
-
-    private void MessLights()
-    {
-        foreach (var comp in EntityQuery<PoweredLightComponent>())
-        {
-            if (RobustRandom.Prob(_event.LightBreakChance))
-            {
-                var uid = comp.Owner;
-                _poweredLight.TryDestroyBulb(uid, comp);
-            }
-        }
-    }
-
     public override void Update(float frameTime)
     {
         base.Update(frameTime);
 
         if (!RuleStarted)
             return;
 
+        foreach (var comp in EntityQuery<PoweredLightComponent>())
+        {
+            if (RobustRandom.Prob(frameTime * _event.LightBreakChancePerSecond))
+                _poweredLight.TryDestroyBulb(comp.Owner, comp);
+        }
+
+        foreach (var comp in EntityQuery<DoorComponent>())
+        {
+            if (RobustRandom.Prob(frameTime * _event.DoorToggleChancePerSecond))
+                _door.TryToggleDoor(comp.Owner, comp);
+        }
+
         if (Elapsed > _event.EndAfter)
         {
```

Comment:
Should this really be checking this every frame?

Reply:
make it check only once a sec? fair